### ISSUE TITLE
ScreenOrientation is not functioning.

### ISSUE THREAD

**Issue by mephisto83** [expo/expo#17498]
### Summary

It worked with sdk version 45.0.0, but there was a bug in the expo-av being built with eas. So, I was advised to upgrade to version 45.0.2, which fixed the expo-av problem, but the ScreenOrientation seems not to function at all.

The ScreenOrientation.addOrientationChangeListener doesn't fire any events.

This won't change the orientation , `ScreenOrientation.lockAsync(ScreenOrientation.OrientationLock.LANDSCAPE_LEFT);`

This doesn't unlock the orientation either.  `await ScreenOrientation.unlockAsync();`

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

^45.0.2

### Environment
```
expo-env-info 1.0.3 environment info:
    System:
      OS: macOS 12.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 18.0.0 - /opt/homebrew/bin/node
      Yarn: 1.22.18 - /opt/homebrew/bin/yarn
      npm: 8.6.0 - /opt/homebrew/bin/npm
    Managers:
      CocoaPods: 1.11.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
    IDEs:
      Android Studio: 2021.1 AI-211.7628.21.2111.8309675
      Xcode: 13.3.1/13E500a - /usr/bin/xcodebuild
    npmPackages:
      expo: ^45.0.2 => 45.0.4 
      react: 17.0.2 => 17.0.2 
      react-dom: 17.0.2 => 17.0.2 
      react-native: 0.68.2 => 0.68.2 
      react-native-web: 0.17.7 => 0.17.7 
    npmGlobalPackages:
      eas-cli: 0.52.0
      expo-cli: 5.4.3
    Expo Workflow: managed
```

### Reproducible demo

```ts
const SomeScreen = ({ route, navigation }: { route: any, navigation: any }) => {

  React.useEffect(() => {
    let res = ScreenOrientation.addOrientationChangeListener(({ orientationInfo, orientationLock }) => {
      switch (orientationInfo.orientation) {
        case ScreenOrientation.Orientation.LANDSCAPE_LEFT:
        case ScreenOrientation.Orientation.LANDSCAPE_RIGHT:
          console.log('orientation changed to lanscape')
          break;
        default:
          console.log('orientation changed to portrait')
          break;
      }
    });
    (async () => {
      console.log('lock left orientation')
      // await ScreenOrientation.unlockAsync();
      await ScreenOrientation.lockAsync(ScreenOrientation.OrientationLock.LANDSCAPE_LEFT);
      console.log('locked left orientation')
    })();
    return () => {
      console.log('lock orientation')
     res.remove();
    }
  }, [])
return <View style={{
    flex: 1,
    justifyContent: 'flex-start',
    backgroundColor:'red'
}}></View>
}
```

**Comment by mephisto83** [comment]
ScreenOrientation does work with expo-go, but not with a eas build. I'm wondering if it is also related to the expo-av issue that was fixed in expo@45.0.2

**Comment by wesvance** [comment]
I can reproduce this using expo init, changing the 'orientation' key value in app.json to 'default' and running a build using eas: eas build --profile local --platform ios --local

The device rotates, but does not update the app with this rotation.

<img width="471" alt="Screen Shot 2022-06-01 at 1 01 40 AM" src="https://user-images.githubusercontent.com/25654206/171332256-5421165f-a327-4066-ac7a-eb32e2a87990.png">

<img width="903" alt="Screen Shot 2022-06-01 at 1 01 20 AM" src="https://user-images.githubusercontent.com/25654206/171332269-9acf1d02-193b-447c-acc0-4134691f325d.png">

**Comment by outaTiME** [comment]
> I can reproduce this using expo init, changing the 'orientation' key value in app.json to 'default' and running a build using eas: eas build --profile local --platform ios --local
> 
> The device rotates, but does not update the app with this rotation.
> 
> ```
> expo-cli: 5.4.4
> expo: 45.0.0
> eas: eas-cli/0.53.0 darwin-arm64 node-v16.14.0
> ```
> 
> <img alt="Screen Shot 2022-06-01 at 1 01 40 AM" width="471" src="https://github.com/assets/25654206/5421165f-a327-4066-ac7a-eb32e2a87990">
> 
> <img alt="Screen Shot 2022-06-01 at 1 01 20 AM" width="903" src="https://github.com/assets/25654206/9acf1d02-193b-447c-acc0-4134691f325d">

Something similar to this issue here: https://github.com/expo/expo/issues/15009#issuecomment-1162002903 ?

**Comment by brentvatne** [comment]
indeed this looks like a duplicate of #15009 / #18005

**Comment by mvanroon** [comment]
@brentvatne #18005 and #15009 are two different issues. #18005 has been fixed, #15009 has not
